pubsub/subscriptions.rb

```diff
@@ -0,0 +1,214 @@
+# Copyright 2016 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This application demonstrates how to perform basic operations on
+# subscriptions with the Google Cloud Pub/Sub API.
+#
+# For more information, see the README.md under /pubsub and the documentation at
+# https://cloud.google.com/pubsub/docs.
+
+require "google/cloud"
+
+gcloud = Google::Cloud.new
+$pubsub_client = gcloud.pubsub
+
+# [START pubsub_list_subscriptions]
+def list_subscriptions
+  # Lists all subscriptions in the current project
+  subscriptions = $pubsub_client.subscriptions
+
+  puts "Subscriptions:"
+  subscriptions.each do |subscription|
+    puts subscription.name
+  end
+end
+# [END pubsub_list_subscriptions]
+
+# [START pubsub_list_topic_subscriptions]
+def list_topic_subscriptions topic_name:
+  # References an existing topic, e.g. "my-topic"
+  topic = $pubsub_client.topic topic_name, skip_lookup: true
+
+  # Lists all subscriptions for the topic
+  subscriptions = topic.subscriptions
+
+  puts "Subscriptions:"
+  subscriptions.each do |subscription|
+    puts subscription.name
+  end
+end
+# [END pubsub_list_topic_subscriptions]
+
+# [START pubsub_create_subscription]
+def create_subscription topic_name:, subscription_name:
+  # References an existing topic, e.g. "my-topic"
+  topic = $pubsub_client.topic topic_name, skip_lookup: true
+
+  # Creates a new subscription, e.g. "my-new-subscription"
+  subscription = topic.subscribe subscription_name
+
+  puts "Subscription #{subscription.name} created."
+end
+# [END pubsub_create_subscription]
+
+# [START pubsub_create_push_subscription]
+def create_push_subscription topic_name:, subscription_name:
+  # References an existing topic, e.g. "my-topic"
+  topic = $pubsub_client.topic topic_name, skip_lookup: true
+
+  project_id = ENV["GCLOUD_PROJECT"] || "YOUR_PROJECT_ID"
```

Comment:
Sample snippets should not contain conditional logic.  This snippet should not include a reference to environment variables.  The project ID should be stored in variable that the developer can edit.

Please use a `project_id` variable such as the following convention followed by Ruby sample snippets:

```ruby
# project_id        = "Your Google Cloud project ID"
# topic_name        = "Name of topic to subscribe to, eg. my-topic"
# subscription_name = "Name of subscription to create, eg. my-subscription"

require "google/cloud"

gcloud = Google::Cloud.new
pubsub = gcloud.pubsub
topic  = pubsub.topic

subscription = topic.subscribe(
  subscription_name,
  # Set to an HTTPS endpoint of your choice. If necessary, register
  # (authorize) the domain on which the server is hosted.
  endpoint: "https://#{project_id}.appspot.com/push"
)

puts "Subscription #{subscription.name} created."
```